plugins/ingest/src/main/java/org/elasticsearch/ingest/Pipeline.java

```diff
@@ -70,6 +70,21 @@ public String getDescription() {
         return processors;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Pipeline pipeline = (Pipeline) o;
+        return Objects.equals(id, pipeline.id) &&
+                Objects.equals(description, pipeline.description) &&
+                Objects.equals(processors, pipeline.processors);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, description, processors);
```

Comment:
just as a side note, I saw the tests that you added, I meant more tests to make sure that equals and hashcode are consistent, I added that to TransportDataTests in my last commit. Also I removed equals/hashcode when it needed to compare exceptions as it wasn't reliable. In that case we simplify our tests in terms of the exception that we can have so we can easily test things.